ember.md

```diff
@@ -0,0 +1,457 @@
+
+3. __Multi line computed property functions__
+
+4. __Lifecycle hooks__
+
+   The hooks should be chronologically ordered by the order they are invoked in.
+
+5. __Functions__
+
+   Public functions first, internal functions after.
+
+6. __Actions__
+
+```js
+export default Component.extend({
+  // Plain properties
+  tagName: 'span',
+
+  // Single line CP
+  post: alias('myPost'),
+
+  // Multiline CP
+  authorName: computed('author.{firstName,lastName}', function _authorName() {
+    // code
+  }),
+
+  // Lifecycle hooks
+  didReceiveAttrs() {
+    this._super(...arguments);
+    // code
+  },
+
+  // Functions
+  someFunction() {
+    // code
+  },
+
+  actions: {
+    someAction() {
+      // Code
+    },
+  },
+});
+```
+
+### Override init
+
+Rather than using the object's `init` hook via `on`, override init and
+call `_super` with `...arguments`. This allows you to control execution
+order. [Don't Don't Override Init](https://dockyard.com/blog/2015/10/19/2015-dont-dont-override-init)
+
+## Models
+
+### Organization
+
+Models should be grouped as follows:
+
+* Attributes
+* Associations
+* Computed Properties
+
+Within each section, the attributes should be ordered alphabetically.
+
+```js
+// Good
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const {
+  attr,
+  hasMany,
+} = DS;
+
+export default DS.Model.extend({
+  // Attributes
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  // Associations
+  children: hasMany('child'),
+
+  // Computed Properties
+  fullName: computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+
+// Bad
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const {
+  attr,
+  hasMany,
+} = DS;
+
+export default DS.Model.extend({
+  children: hasMany('child'),
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  fullName: computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+
+```
+
+## Controllers
+
+### Define query params first
+
+For consistency and ease of discover, list your query params first in
+your controller. These should be listed above default values.
+
+### Alias your model
+
+It provides a cleaner code to name your model `user` if it is a user. It
+is more maintainable, and will fall in line with future routable
+components
+
+```javascript
+export default Ember.Controller.extend({
+  user: computed.readOnly('model'),
+});
+```
+
+## Templates
+
+### Avoid using partials
+
+Use components. Partials share scope with the parent view, which makes it hard
+to follow the context. Components will provide a consistent scope. Only use
+partials if you really, really need the shared scope.
+
+### Don't yield `this`
+
+Use the hash helper to yield what you need instead.
+
+```hbs
+{{! Good }}
+{{yield (hash thing=thing action=(action "action"))}}
+
+{{! Bad }}
+{{yield this}}
+```
+
+### Use components in `{{#each}}` blocks
```

Comment:
I think this is a good rule-of-thumb, but would be happy to be somewhat more relaxed about it. I didn't edit the language here, but it would be changed to be slightly more relaxed?